### PR TITLE
fix: [TPM] correcting cfg data and key hash

### PR DESCRIPTION
**Pull request by aborwank**
issue: https://jira.devtools.intel.com/browse/IOTGSECSW-2031

### CODE REVIEW

File: BootloaderCorePkg/Stage1B/Stage1B.c [hunk 1]
```diff
@@ -731,8 +730,8 @@ ContinueFunc (
                     MbTmpAlgHash,
                     Stage1bParam->ConfigDataHash,
                     EV_PLATFORM_CONFIG_FLAGS,
-                    sizeof(CfgDataFwBlob),
-                    (UINT8 *)&CfgDataFwBlob);
+                    (UINT32)CfgDataFwBlob.BlobLength,
```
Comment (gdong1) [inline BootloaderCorePkg/Stage1B/Stage1B.c:733]: How about directly use CfgDataBlob->UsedLength and CfgDataFwBlob could be removed?

File: BootloaderCorePkg/Stage1B/Stage1B.c [hunk 2]
```diff
@@ -744,8 +743,8 @@ ContinueFunc (
                     MbTmpAlgHash,
                     Stage1bParam->KeyHashManifestHash,
                     EV_EFI_PLATFORM_FIRMWARE_BLOB,
-                    sizeof(KeyHashFwBlob),
-                    (UINT8 *)&KeyHashFwBlob);
+                    (UINT32)KeyHashFwBlob.BlobLength,
```
Comment (gdong1) [inline BootloaderCorePkg/Stage1B/Stage1B.c:746]: How about directly use KeyHashBlob->UsedLength and remove variable KeyHashFwBlob?

**Comment by gdong1** [comment]
Good catch the issue and have a fix. It looks there is another place (https://github.com/slimbootloader/slimbootloader/blob/2648eb75f839d9c88607117c2565c04ad0ae510f/BootloaderCommonPkg/Library/TpmLib/TpmLib.c#L1015) that has similar issue.  Could you help fix it in this PR?

File: BootloaderCorePkg/Stage1B/Stage1B.c (hunk 1 above)
Comment (aborwank) [reply BootloaderCorePkg/Stage1B/Stage1B.c:733]: done

File: BootloaderCorePkg/Stage1B/Stage1B.c (hunk 2 above)
Comment (aborwank) [reply BootloaderCorePkg/Stage1B/Stage1B.c:746]: done

**Comment by aborwank** [comment]
> Good catch the issue and have a fix. It looks there is another place (
> 
> https://github.com/slimbootloader/slimbootloader/blob/2648eb75f839d9c88607117c2565c04ad0ae510f/BootloaderCommonPkg/Library/TpmLib/TpmLib.c#L1015
> 
> ) that has similar issue. Could you help fix it in this PR?

done

**Review by lsubash — COMMENTED** [review]
Physical SPI address and blob length has to be captured in the TPM event log which would be per the flash map. Please check the TPM event log on Linux for the firmware blobs.
From the container, CbInfo would contain the Ram address.

**Pull request merged**